In [1]:
import numpy as np
import pandas as pd
import sqlite3
import os

In [2]:
v13_path = '/Users/yoachim/Temp/sims_featureScheduler_runs_1.3/maf'
v13_dirs = ['baseline_2snap_v1.3_10yrs',
            'baseline_nomix_v1.3_10yrs',
            'baseline_v1.3_10yrs',
            'baseline_v1.3_1yrs',
           'wfd_95_v1.3_10yrs',
           'wfd_80_v1.3_10yrs',
           'wfd_65_v1.3_10yrs',
           'delayedrolling_mod2_sdf_0.10_v1.3_10yrs']

dirs = [os.path.join(v13_path, filename) for filename in v13_dirs]

dirs.append('/Users/yoachim/Temp/old_runs/baseline2018a_note')

In [3]:
v12_path = '/Users/yoachim/gitRepos/sims_featureScheduler_runs/baselines'
v12_dirs = ['baseline_1exp_pairsame_10yrs',
           'baseline_1exp_pairsmix_10yrs',
           'baseline_2exp_pairsame_10yrs',
           'baseline_2exp_pairsmix_10yrs']

for directory in v12_dirs:
    dirs.append(os.path.join(v12_path, directory))

In [4]:
results = {'fONv MedianNvis':[], 'fOArea': [], 'osf':[], 'nexp':[]}
for directory in dirs:
    conn = sqlite3.connect(os.path.join(directory, 'resultsDb_sqlite.db'))
    df = pd.read_sql('select summaryValue from summarystats where summaryName="fONv MedianNvis";', conn)
    results['fONv MedianNvis'].append(np.max(df.values))
    
    df = pd.read_sql('select summaryValue from summarystats where summaryName="fOArea";', conn)
    results['fOArea'].append(np.max(df.values))
    
    df = pd.read_sql('select summaryValue from summarystats where metricid=3;', conn)
    results['osf'].append(np.max(df.values))
    
    df = pd.read_sql('select summaryValue from summarystats where metricid=5;', conn)
    results['nexp'].append(np.max(df.values))
    
    conn.close()

In [5]:
results


{'fOArea': [10052.219988694145,
  18127.06414927078,
  18090.135228881489,
  67.982785262104514,
  18404.031052190465,
  18212.67210108232,
  10551.599707594789,
  16123.670218151725,
  18056.563483073041,
  18226.940093050907,
  18186.653998080772,
  18096.849578043177,
  17942.419547324324],
 'fONv MedianNvis': [828.0,
  915.0,
  900.0,
  94.0,
  1040.0,
  922.0,
  831.0,
  880.0,
  915.0,
  986.0,
  963.0,
  907.0,
  887.0],
 'nexp': [2019329.0,
  2230670.0,
  2182491.0,
  228046.0,
  2190893.0,
  2190844.0,
  2194390.0,
  2192506.0,
  2372700.0,
  2263993.0,
  2216026.0,
  2092909.0,
  2048865.0],
 'osf': [0.7027261183796657,
  0.77623484079066107,
  0.75941748316485058,
  0.76720785006963166,
  0.76235593935538637,
  0.7623543785010467,
  0.76357136217036081,
  0.76291726080211031,
  0.71565104665181434,
  0.78456252309099028,
  0.7679721152855904,
  0.72535594217231303,
  0.71010851612466019]}

In [14]:
keys = list(results.keys())
header = '| Run Name  |    '
for key in keys:
    header+= '%s |' % key

print(header)
print('| -- | -- | -- | -- | --|')
for i in range(len(dirs)):
    rn = os.path.basename(dirs[i])
    out = '| '
    out += rn +' |'
    for key in keys:
        out +='   %.3f  |' % results[key][i]
    print(out)
    

| Run Name  |    fONv MedianNvis |fOArea |osf |nexp |
| -- | -- | -- | -- | --|
| baseline_2snap_v1.3_10yrs |   828.000  |   10052.220  |   0.703  |   2019329.000  |
| baseline_nomix_v1.3_10yrs |   915.000  |   18127.064  |   0.776  |   2230670.000  |
| baseline_v1.3_10yrs |   900.000  |   18090.135  |   0.759  |   2182491.000  |
| baseline_v1.3_1yrs |   94.000  |   67.983  |   0.767  |   228046.000  |
| wfd_95_v1.3_10yrs |   1040.000  |   18404.031  |   0.762  |   2190893.000  |
| wfd_80_v1.3_10yrs |   922.000  |   18212.672  |   0.762  |   2190844.000  |
| wfd_65_v1.3_10yrs |   831.000  |   10551.600  |   0.764  |   2194390.000  |
| delayedrolling_mod2_sdf_0.10_v1.3_10yrs |   880.000  |   16123.670  |   0.763  |   2192506.000  |
| baseline2018a_note |   915.000  |   18056.563  |   0.716  |   2372700.000  |
| baseline_1exp_pairsame_10yrs |   986.000  |   18226.940  |   0.785  |   2263993.000  |
| baseline_1exp_pairsmix_10yrs |   963.000  |   18186.654  |   0.768  |   2216026.000  |
| 

| Run Name  |    fONv MedianNvis |fOArea |osf |nexp |
| -- | -- | -- | -- | --|
| baseline_2snap_v1.3_10yrs |   828.000  |   10052.220  |   0.703  |   2019329.000  |
| baseline_nomix_v1.3_10yrs |   915.000  |   18127.064  |   0.776  |   2230670.000  |
| baseline_v1.3_10yrs |   900.000  |   18090.135  |   0.759  |   2182491.000  |
| baseline_v1.3_1yrs |   94.000  |   67.983  |   0.767  |   228046.000  |
| wfd_95_v1.3_10yrs |   1040.000  |   18404.031  |   0.762  |   2190893.000  |
| wfd_80_v1.3_10yrs |   922.000  |   18212.672  |   0.762  |   2190844.000  |
| wfd_65_v1.3_10yrs |   831.000  |   10551.600  |   0.764  |   2194390.000  |
| delayedrolling_mod2_sdf_0.10_v1.3_10yrs |   880.000  |   16123.670  |   0.763  |   2192506.000  |
| baseline2018a_note |   915.000  |   18056.563  |   0.716  |   2372700.000  |
| baseline_1exp_pairsame_10yrs |   986.000  |   18226.940  |   0.785  |   2263993.000  |
| baseline_1exp_pairsmix_10yrs |   963.000  |   18186.654  |   0.768  |   2216026.000  |
| baseline_2exp_pairsame_10yrs |   907.000  |   18096.850  |   0.725  |   2092909.000  |
| baseline_2exp_pairsmix_10yrs |   887.000  |   17942.420  |   0.710  |   2048865.000  |

| Run Name  | version | Pairs | snaps |  fONv MedianNvis |fOArea |osf |nexp |
| -- | -- | -- | -- | --|  -- | -- | --|
| baseline_2snap_v1.3_10yrs | FBS1.3 | Mixed  |  2 | 828.000  |   10052.220  |   0.703  |   2019329.000  |
| baseline_nomix_v1.3_10yrs |  FBS1.3   | Same  | 1  |915.000  |   18127.064  |   0.776  |   2230670.000  |
| baseline_v1.3_10yrs |  FBS1.3  | Mixed  |  1 | 900.000  |   18090.135  |   0.759  |   2182491.000  |
| baseline_v1.3_1yrs |  FBS1.3   | Mixed  |  1 |94.000  |   67.983  |   0.767  |   228046.000  |
| wfd_95_v1.3_10yrs | FBS1.3   |  Mixed | 1  | 1040.000  |   18404.031  |   0.762  |   2190893.000  |
| wfd_80_v1.3_10yrs |  FBS1.3   | Mixed  | 1  |922.000  |   18212.672  |   0.762  |   2190844.000  |
| wfd_65_v1.3_10yrs |  FBS1.3   | Mixed  |  1 |831.000  |   10551.600  |   0.764  |   2194390.000  |
| delayedrolling_mod2_sdf_0.10_v1.3_10yrs |  FBS1.3  | Mixed  |  1 | 880.000  |   16123.670  |   0.763  |   2192506.000  |
| baseline2018a_note |  OpSimv4  | Same  | 2  | 915.000  |   18056.563  |   0.716  |   2372700.000  |
| baseline_1exp_pairsame_10yrs | FBS1.2   | Same  |  1 | 986.000  |   18226.940  |   0.785  |   2263993.000  |
| baseline_1exp_pairsmix_10yrs |  FBS1.2  | Mixed  |  1 | 963.000  |   18186.654  |   0.768  |   2216026.000  |
| baseline_2exp_pairsame_10yrs | FBS1.2   | Same  |  2 | 907.000  |   18096.850  |   0.725  |   2092909.000  |
| baseline_2exp_pairsmix_10yrs |  FBS1.2  | Mixed  |  2 | 887.000  |   17942.420  |   0.710  |   2048865.000  |